### PR TITLE
Convert jsf javax tests from groovy to java

### PR DESCRIPTION
**Pull request by jaydeluca**
Related to #7195

### CODE REVIEW

File: instrumentation/jsf/jsf-javax-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/jsf/javax/BaseJsfTest.java [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+
+  @Override
+  protected Server setupServer() throws Exception {
+    List<String> configurationClasses = new ArrayList<>();
+    addAll(configurationClasses, WebAppContext.getDefaultConfigurationClasses());
+    configurationClasses.add(AnnotationConfiguration.class.getName());
+
+    WebAppContext webAppContext = new WebAppContext();
+    webAppContext.setContextPath(getContextPath());
+    // set up test application
```
Comment (steverao) [inline instrumentation/jsf/jsf-javax-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/jsf/javax/BaseJsfTest.java:81]: It seems you miss below line above this line:
```java
webAppContext.setConfigurationClasses(configurationClasses);
```

Comment (jaydeluca) [reply instrumentation/jsf/jsf-javax-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/jsf/javax/BaseJsfTest.java:82]: thanks!

**Review by steverao — APPROVED** [review]

File: instrumentation/jsf/jsf-javax-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/jsf/javax/BaseJsfTest.java [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+                span ->
+                    span.hasName(spanName)
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(parentIndex))));
+
+    if (expectedException != null) {
+      assertions.add(
+          span ->
+              span.hasStatus(StatusData.error())
+                  .hasEventsSatisfyingExactly(
```
Comment (laurit) [inline instrumentation/jsf/jsf-javax-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/jsf/javax/BaseJsfTest.java:232]: Could this be replaced with `.hasException(expectedException)`?

File: instrumentation/jsf/jsf-javax-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/jsf/javax/BaseJsfTest.java [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName(getContextPath() + "/greeting.xhtml")
+                        .hasKind(SpanKind.SERVER)
+                        .hasNoParent()
+                        .hasStatus(StatusData.error())
+                        .hasEventsSatisfyingExactly(
```
Comment (laurit) [inline instrumentation/jsf/jsf-javax-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/jsf/javax/BaseJsfTest.java:310]: Could this be replaced with .hasException(ex)?

**Comment by laurit** [comment]
Seems that to use `hasException` we need to restore unwrapping the exception that the groovy version had https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11736

**Review by laurit — APPROVED** [review]

**Pull request merged**